### DISCUSSION TITLE
Docker Container access

### DISCUSSION THREAD

**Discussion by monicamillad** [etcd-io/etcd#16806] [Q&A] [answered]
Hello, 
I was trying to access the container after running it but I didn't succeed

the command I used to run was 

`docker run -d --rm --ip 10.20.30.10 --volume /etcd-vol --volume /etcd_backup --name etcd gcr.io/etcd-development/etcd:v3.4.27 /usr/local/bin/etcd --data-dir=/etcd-data/etcd-data --name etcd-node-0 --initial-advertise-peer-urls http://10.20.30.10:2380 --listen-peer-urls http://0.0.0.0:2380 --advertise-client-urls http://10.20.30.10:2379 --listen-client-urls http://0.0.0.0:2379`

and it ran successfully, but after that I was trying something like `docker exec -it etcd bash` or `docker exec -it etcd sh` but I always get this error `OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "bash": executable file not found in $PATH: unknown`

so I was wondering if I doing something wrong or I missed something ,, or maybe the container doesn't have bash or sh I don't know

**Comment by jmhbnz** [comment, marked as answer]
Hey @monicamillad - Thanks for your question.

To minimise attack surface the standard etcd images are based on distroless and do not contain a shell, refer: https://github.com/etcd-io/etcd/blob/245b58a1e280770cc9132a92b1ab34bc722e4dd8/Dockerfile#L2 

If you need to interact with etcd you can `docker exec` to run the `etcdctl` utility available inside the container. For example:

```bash
# Check etcd endpoint status
docker exec <name> etcdctl endpoint status -w table
```

**Reply by monicamillad** [reply]
this worked for me thank you very much
